### PR TITLE
Add native Maltiverse integration

### PR DESCRIPTION
**Pull request by dgilm**
|Related issue|
|#17752|
|#17753|

## Description

This PR adds the Maltiverse native integration to Wazuh.

### What is Maltiverse?
Maltiverse works as a broker for Threat intelligence sources that are
aggregated from more than a hundred different Public, Private and Community
sources. Once the data is ingested, the IoC Scoring Algorithm applies a
qualitative classification to the IoC that changes. Finally this data can
be queried in a Threat Intelligence feed that can be delivered to your
Firewalls, SOAR, SIEM, EDR or any other technology.

More info at: https://whatis.maltiverse.com

### What does this integration?
This integration enrichs any alert generated by Wazuh via the Maltiverse API,
inserting new fields in case of match and following the threat taxonomy of the
ECS standard (Elastic Common Squema).

https://www.elastic.co/guide/en/ecs/current/ecs-threat.html

Ipv4, Domain names, Urls and MD5/SHA1 checksums are checked in Maltiverse
platform in order to enrich the original alert with threat Intel information

## Configuration options

1. Move this file ``maltiverse.py`` to ``/var/ossec/integrations/custom-maltiverse.py``
   and make sure the file has the right perms:
   ```
   chmod +x /var/ossec/integrations/custom-maltiverse.py
   chown root.wazuh /var/ossec/integrations/custom-maltiverse.py
    ```

2. Add this to the ossec.conf file, inside <ossec_config></ossec_config> block:
```
<integration>
     <name>custom-maltiverse</name>
     <hook_url>https://api.maltiverse.com</hook_url>
     <api_key><YOUR_MALTIVERSE_AUTH_TOKEN></api_key>
     <alert_format>json</alert_format>
</integration>
```
3. Restart Wazuh Manager

## Logs/Alerts example

Threat taxonomy of the ECS standard (Elastic Common Squema) is followed: https://www.elastic.co/guide/en/ecs/current/ecs-threat.html

```
      "threat": {
        "indicator": {
          "reference": "https://maltiverse.com/ip/212.192.241.132",
          "first_seen": "2021-12-30 00:58:57",
          "last_seen": "2023-07-05 13:59:09",
          "provider": "Emerging Threats, Maltiverse, Blocklist.de, FireHOL",
          "confidence": "Medium",
          "name": "212.192.241.132",
          "description": "Scanning IPs, ET COMPROMISED Known Compromised or Hostile Host Traffic TCP, HTTP Attacker, Bruteforce login attacker, Proxy, ET COMPROMISED Known Compromised or Hostile Host Traffic UDP",
          "sightings": "6",
          "type": "ip",
          "modified_at": "2023-07-05 13:59:09"
        }
      }
```

Also, full response of Maltiverse api is included in `data->maltiverse->source`

## Tests

- Compilation without warnings in every supported platform
  - [ ] Linux
  - [ ] Windows
  - [ ] MAC OS X
- [ ] Source installation
- [ ] Package installation
- [ ] Source upgrade
- [ ] Package upgrade
- [ ] Review logs syntax and correct language
- [ ] QA templates contemplate the added capabilities

- Memory tests for Linux
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Valgrind (memcheck and descriptor leaks check)
  - [ ] Dr. Memory
  - [ ] AddressSanitizer
- Memory tests for Windows
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Dr. Memory
- Memory tests for macOS
  - [ ] Scan-build report
  - [ ] Leaks
  - [ ] AddressSanitizer

- [ ] Retrocompatibility with older Wazuh versions
- [ ] Working on cluster environments
- [ ] Configuration on demand reports new parameters
- [ ] The data flow works as expected (agent-manager-api-app)
- [ ] Added unit tests (for new features)
- [ ] Stress test for affected components

- Decoder/Rule tests
  - [ ] Added unit testing files ".ini"
  - [ ] runtests.py executed without errors

### CODE REVIEW

File: ruleset/rules/0995-maltiverse_rules.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<group name="maltiverse,">
+    <rule id="99500" level="0">
```
Comment (fdalmaup) [inline ruleset/rules/0995-maltiverse_rules.xml:2]:
```suggestion
    <rule id="99500" level="10">
```
We should define a base level, e.g. 10, for the rule to trigger alerts when Maltiverse returns aggregated information.

**Review by fdalmaup — CHANGES_REQUESTED** [review]
Other general suggestions:
- I have tested the integration and works fine but test output should be added here or in the development issue.
- We should change the target branch to the epic branch [17752-maltiverse-integration](https://github.com/wazuh/wazuh/tree/17752-maltiverse-integration).

File: integrations/maltiverse.py [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+    This is just a template used to build the generated alert using
+    the following arguments:
+
+    alert_id: the generated alert id
+    ioc_dict: raw information returned by Maltiverse API
+    ioc_name: the representative name of the indicator
+    ioc_ref: (optional) an indicator reference used to build a reference url
+      ioc_name is used by dafault if ioc_ref is not set
+    include_full_source: (optional) if True (default), the complete API
+      response is also included.
```
Comment (fdalmaup) [inline integrations/maltiverse.py:278]:
```suggestion
    alert_id: in
        The generated alert ID.
    ioc_dict: dict 
        Raw information returned by Maltiverse API.
    ioc_name: str
        The representative name of the indicator.
    ioc_ref: Optional[str] 
        An indicator reference used to build a reference URL.
        ioc_name is used by default if ioc_ref is not set.
    include_full_source: Optional[bool] 
        If True (default), the complete API response is also included.
```

**Review by fdalmaup — CHANGES_REQUESTED** [review]

**Review by fdalmaup — APPROVED** [review]
LGTM!

**Review by Selutario — APPROVED** [review]
LGTM

**Pull request merged**